packages/stacked_themes/lib/src/theme_builder.dart

```diff
@@ -58,4 +60,62 @@ class _ThemeBuilderState extends State<ThemeBuilder> {
       ),
     );
   }
+
+  // Get all services
+  // final themeService = locator<ThemeService>();
+  // @override
+  // Widget build(BuildContext context) {
+  //   return widget.child;
+  // }
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    WidgetsBinding.instance.removeObserver(this);
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    super.didChangeAppLifecycleState(state);
+
+    switch (state) {
+      case AppLifecycleState.inactive:
+        print('appLifeCycleState inactive');
+        break;
+      case AppLifecycleState.resumed:
+        print('appLifeCycleState resumed');
+        adjustSystemThemeIfNecessary();
+        break;
+      case AppLifecycleState.paused:
+        print('appLifeCycleState paused');
+        break;
+      case AppLifecycleState.detached:
+        print('appLifeCycleState suspending');
+        break;
+    }
+  }
+
+  //NOTE: re-apply the appropriate theme when the application gets back into the foreground
+  void adjustSystemThemeIfNecessary() {
+    switch (themeManager.selectedThemeMode) {
+      //do nothing
+      case ThemeMode.dark:
+        break;
+      //do nothing
+      case ThemeMode.dark:
+        break;
+      //reapply theme
+      case ThemeMode.system:
+        print('reApplying system theme');

```

Comment:
Please remove the log or make it optional.

Reply:
thanks for the feedback I'll update the pull request